### ISSUE TITLE
libpl_droidsonroids_gif.so1.2.7" not found

### ISSUE THREAD

**Issue by MrApocalypse** [koral--/android-gif-drawable#426]
Hello, this is a great library that makes everything with GIFs better, i just have one issue. Loading a GifImageView crashes my app when opened on a real phone (Nexus 6p, Android 7.1.2_r27), however works fine & loads when ran from the emulator (Android Emulator, Android 7.1.1, API 25). Very weird, not sure what's going on. The app's data has been deleted and performed fresh install, here's my log.
`java.lang.RuntimeException: Unable to start activity ComponentInfo{com.screwdaosp.lean/com.screwdaosp.lean.MainActivity}: android.view.InflateException: Binary XML file line #53: Binary XML file line #53: Error inflating class pl.droidsonroids.gif.GifImageView
                                                       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2681)
                                                       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2742)
                                                       at android.app.ActivityThread.-wrap12(ActivityThread.java)
                                                       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1489)
                                                       at android.os.Handler.dispatchMessage(Handler.java:102)
                                                       at android.os.Looper.loop(Looper.java:154)
                                                       at android.app.ActivityThread.main(ActivityThread.java:6171)
                                                       at java.lang.reflect.Method.invoke(Native Method)
                                                       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:891)
                                                       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:781)
                                                    Caused by: android.view.InflateException: Binary XML file line #53: Binary XML file line #53: Error inflating class pl.droidsonroids.gif.GifImageView
                                                    Caused by: android.view.InflateException: Binary XML file line #53: Error inflating class pl.droidsonroids.gif.GifImageView
                                                    Caused by: java.lang.reflect.InvocationTargetException
                                                       at java.lang.reflect.Constructor.newInstance0(Native Method)
                                                       at java.lang.reflect.Constructor.newInstance(Constructor.java:430)
                                                       at android.view.LayoutInflater.createView(LayoutInflater.java:645)
                                                       at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:787)
                                                       at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:727)
                                                       at android.view.LayoutInflater.rInflate(LayoutInflater.java:858)
                                                       at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:821)
                                                       at android.view.LayoutInflater.rInflate(LayoutInflater.java:861)
                                                       at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:821)
                                                       at android.view.LayoutInflater.rInflate(LayoutInflater.java:861)
                                                       at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:821)
                                                       at android.view.LayoutInflater.inflate(LayoutInflater.java:518)
                                                       at android.view.LayoutInflater.inflate(LayoutInflater.java:426)
                                                       at android.view.LayoutInflater.inflate(LayoutInflater.java:377)
                                                       at android.support.v7.app.AppCompatDelegateImplV9.setContentView(AppCompatDelegateImplV9.java:292)
                                                       at android.support.v7.app.AppCompatActivity.setContentView(AppCompatActivity.java:140)
                                                       at com.screwdaosp.lean.MainActivity.onCreate(MainActivity.java:99)
                                                       at android.app.Activity.performCreate(Activity.java:6687)
                                                       at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1118)
                                                       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2634)
                                                       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2742)
                                                       at android.app.ActivityThread.-wrap12(ActivityThread.java)
                                                       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1489)
                                                       at android.os.Handler.dispatchMessage(Handler.java:102)
                                                       at android.os.Looper.loop(Looper.java:154)
                                                       at android.app.ActivityThread.main(ActivityThread.java:6171)
                                                       at java.lang.reflect.Method.invoke(Native Method)
                                                       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:891)
                                                       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:781)
                                                    Caused by: java.lang.UnsatisfiedLinkError: dlopen failed: library "/data/user/0/com.screwdaosp.lean/cache/libpl_droidsonroids_gif.so1.2.7" not found
                                                       at java.lang.Runtime.load0(Runtime.java:908)
                                                       at java.lang.System.load(System.java:1537)
                                                       at pl.droidsonroids.gif.ReLinker.loadLibrary(ReLinker.java:59)
                                                       at pl.droidsonroids.gif.LibraryLoader.loadLibrary(LibraryLoader.java:52)
                                                       at pl.droidsonroids.gif.GifInfoHandle.<clinit>(GifInfoHandle.java:21)
                                                       at pl.droidsonroids.gif.GifDrawable.<init>(GifDrawable.java:160)
                                                       at pl.droidsonroids.gif.GifDrawable.<init>(GifDrawable.java:95)
                                                       at pl.droidsonroids.gif.GifViewUtils.setResource(GifViewUtils.java:53)
07-21 17:16:23.115 10278-10278/? E/AndroidRuntime:     at pl.droidsonroids.gif.GifViewUtils.getResourceId(GifViewUtils.java:41)
                                                       at pl.droidsonroids.gif.GifViewUtils.initImageView(GifViewUtils.java:29)
                                                       at pl.droidsonroids.gif.GifImageView.<init>(GifImageView.java:41)`

**Comment by koral--** [comment]
Does it help if you add `android:extractNativeLibs="false"` to `<application>` tag in manifest?
Reference: https://developer.android.com/guide/topics/manifest/application-element.html#extractNativeLibs

**Comment by MrApocalypse** [comment]
Yes, that does help, in fact it fixes my issue. Thank you!

**Comment by koral--** [comment]
OK, I'll add this info to readme.
TL;DR loading native libraries is buggy in Android, that's why Relinker is here. More info: https://medium.com/keepsafe-engineering/the-perils-of-loading-native-libraries-on-android-befa49dce2db
This issue shows that even that workaround is not sufficient.

Starting from API 23+ there is a new attribute `android:extractNativeLibs` (ignored on older API levels). If it is set to `false` native libs are loaded directly from APK which should be less error-prone.
